### PR TITLE
Add base config layer

### PR DESCRIPTION
**Pull request by aahmed-se**

### CODE REVIEW

File: core/src/main/java/org/openconnectors/Config.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.streamlio.config;
```
Comment (jerrypeng) [inline core/src/main/java/org/openconnectors/Config.java]: I think this is not correct. Probably should be org.openconnectors

Comment (jerrypeng) [reply core/src/main/java/org/openconnectors/Config.java]: is a problem for all files i think

Comment (aahmed-se) [reply core/src/main/java/org/openconnectors/Config.java]: fixed

File: core/src/main/java/org/openconnectors/Config.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.openconnectors.config;
```
Comment (Antwnis) [inline core/src/main/java/org/openconnectors/Config.java:20]: refactor files
`.java` files should be under a `java/org/openconnectors/config/` folder structure

Comment (aahmed-se) [reply core/src/main/java/org/openconnectors/Config.java:20]: done

**Comment by aahmed-se** [comment]
Don't want to take a hard dependency on typesafe, I want to keep the components separated at the Config.java interface which is generic to any implementation.

**Pull request merged**